action/protocol/execution/evm/evm.go

```diff
@@ -423,19 +425,40 @@ func getChainConfig(g genesis.Blockchain, height uint64, id uint32, getBlockTime
 	sumatraTime, err := getBlockTime(g.SumatraBlockHeight)
 	if err != nil {
 		return nil, err
+	} else if sumatraTime != nil {
+		sumatraTimestamp := (uint64)(sumatraTime.Unix())
+		chainConfig.ShanghaiTime = &sumatraTimestamp
 	}
-	sumatraTimestamp := (uint64)(sumatraTime.Unix())
-	chainConfig.ShanghaiTime = &sumatraTimestamp
 	// enable Cancun at Vanuatu
 	cancunTime, err := getBlockTime(g.VanuatuBlockHeight)
 	if err != nil {
 		return nil, err
+	} else if cancunTime != nil {
+		cancunTimestamp := (uint64)(cancunTime.Unix())
+		chainConfig.CancunTime = &cancunTimestamp
 	}
-	cancunTimestamp := (uint64)(cancunTime.Unix())
-	chainConfig.CancunTime = &cancunTimestamp
 	return &chainConfig, nil
 }
 
+// blockHeightToTime returns the block time by height
+// if height is greater than current block height, return nil
+// if height is equal to current block height, return current block time
+// otherwise, return the block time by height from the blockchain
+func blockHeightToTime(ctx context.Context, height uint64) (*time.Time, error) {
+	blkCtx := protocol.MustGetBlockCtx(ctx)
+	if height > blkCtx.BlockHeight {
+		return nil, nil
+	}
+	if height == blkCtx.BlockHeight {
+		return &blkCtx.BlockTimeStamp, nil
+	}
+	t, err := mustGetHelperCtx(ctx).GetBlockTime(height)
```

Comment:
can you check if `t == nil` when this func returns `err != nil`? if that's the case we can directly do
```
return mustGetHelperCtx(ctx).GetBlockTime(height)
```

Reply:
discussed offline